providers/edge3/src/airflow/providers/edge3/plugins/www/package.json

```diff
@@ -0,0 +1,80 @@
+{
+  "name": "edge",
+  "private": true,
+  "version": "0.0.0",
+  "type": "module",
+  "main": "./dist/main.js",
+  "module": "./dist/main.js",
+  "types": "./dist/main.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/main.js",
+      "types": "./dist/main.d.ts"
+    }
+  },
+  "files": [
+    "dist"
+  ],
+  "scripts": {
+    "dev": "vite --port 5173 --strictPort",
+    "build": "vite build",
+    "build:types": "tsc --p tsconfig.lib.json",
+    "build:lib": "vite build",
+    "lint": "eslint --quiet && tsc --p tsconfig.app.json",
+    "lint:fix": "eslint --fix && tsc --p tsconfig.app.json",
+    "format": "pnpm prettier --write .",
+    "preview": "vite preview",
+    "codegen": "openapi-rq -i ../../openapi/v2-edge-generated.yaml -c axios --format prettier -o openapi-gen --operationId",
```

Comment:
Generating API hooks based on the plugin endpoints? Cool!

Reply:
It's done as part of the precommit hook. `ts-compile-lint-ui`. This way the generated code and source spec file are always in sync.